x-pack/platform/plugins/shared/onechat/public/application/components/tools/form/components/index_search/index_search_pattern.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  EuiBadge,
+  EuiBasicTable,
+  type EuiBasicTableColumn,
+  EuiSpacer,
+  EuiFieldText,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useController, useFormContext } from 'react-hook-form';
+import { labels } from '../../../../../utils/i18n';
+import { useIndexSearchSources } from '../../../../../hooks/tools/use_resolve_search_sources';
+import type { IndexSearchToolFormData } from '../../types/tool_form_types';
+
+const DEFAULT_PAGE_SIZE = 5;
+
+export const IndexSearchPattern: React.FC = () => {
+  const { control, setValue, setError, clearErrors } = useFormContext<IndexSearchToolFormData>();
+  const {
+    field: { ref, value, onChange, onBlur, name },
+    fieldState,
+  } = useController({ name: 'pattern', control });
+
+  const patternValue = value ?? '';
+  const hasQuery = (patternValue.trim().length ?? 0) > 0;
```

Comment:
nit: nullish coalescing not needed here since above line takes care of it


```suggestion
  const hasQuery = (patternValue.trim().length) > 0;
```